d2/src/main/java/com/linkedin/d2/balancer/clusterfailout/ClusterFailoutBucketConfig.java

```diff
@@ -0,0 +1,129 @@
+package com.linkedin.d2.balancer.clusterfailout;
+
+import java.util.Map;
+import java.util.Date;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is a simple data structure for tracking the traffic state of a given application.
+ */
+
+
+/*
+failedoutBuckets: array[BucketFailoutConfig] {
+    fabric: FabricUrn
+    partition: string
+    // e.g. buckets 0-4 are offline
+    bucketIdRange: IntRange
+    offlineAt: Time
+    onlineAt: optional Time
+  }
+ */
+public class ClusterFailoutBucketConfig
+{
+  // Fabric that this configuration is addressing.
+  private final String _fabricUrn;
+  // Traffic partition, eg. Main, Guest, Static, etc.
+  private final String _partition;
+  // Range of buckets offline. Eg, 5, so buckets 0-4 are offline.
+  private final Integer _bucketOfflineRange;
```

Comment:
Why do we use a single integer to specify a range? Seems more natural to me by translating to bucketOfflineRangeStart and bucketOfflineRangeEnd.

Reply:
The range is 0 - integer, inclusive. There's no advantage to failing out subsets of buckets in the middle of the superset, so a single integer works here.

Reply:
so the var name bucketOffline**Range** is a bit misleading. Can it be bucketOfflineRange**End** or buckOffline**Bound**?